### DISCUSSION TITLE
Is it possible to 'abstract' the urql client init in next-urql?

### DISCUSSION THREAD

**Discussion by luisfontes** [urql-graphql/urql#2036] [Q&A] [answered]
Hi! I've seen the [next-urql examples](https://github.com/FormidableLabs/urql/blob/main/examples/with-next/pages/static.js) on how to handle SSG/ISG and have the following question:

Is it possible to 'abstract' the urql client so that you don't have always to import the client, all the exchanges and repeat the config in every page?

I've found some examples on github but I don't know if it's the correct way of doing it:

Example 1:
[https://github.com/AymerichDelettre/tsReactNextGraphQlBoilerplate/blob/master/src/utils/createUrqlClient.ts](https://github.com/AymerichDelettre/tsReactNextGraphQlBoilerplate/blob/master/src/utils/createUrqlClient.ts)

Example 2:
![image](https://user-images.githubusercontent.com/5988029/138302835-432523f6-9c6f-4a35-9e62-7c2063457fed.png)

**Comment by JoviDeCroock** [comment, marked as answer]
Yes, you can export a function that creates your client and export it from a different file

**Reply by luisfontes** [reply]
Thanks for the answer, I got confused if it was possible because of this discussion: https://github.com/FormidableLabs/urql/discussions/1521
